java/vector/src/main/java/org/apache/arrow/vector/ipc/ArrowWriter.java

```diff
@@ -77,23 +78,32 @@ protected ArrowWriter(VectorSchemaRoot root, DictionaryProvider provider, Writab
    * @param option   IPC write options
    */
   protected ArrowWriter(VectorSchemaRoot root, DictionaryProvider provider, WritableByteChannel out, IpcOption option) {
-    this.unloader = new VectorUnloader(root);
+    this.root = root;
+    this.unloader = new VectorUnloader(this.root);
     this.out = new WriteChannel(out);
     this.option = option;
+    this.provider = provider;
+
+    initialize();
+  }
 
-    List<Field> fields = new ArrayList<>(root.getSchema().getFields().size());
+  /**
+   * Initialize object needed as part of Arrow Writer functionalities.
+   */
+  public void initialize() {
```

Comment:
As noted above, I'd rather not have this, but if we do  have to have it, it should be protected at most.